golang/src/com/google/idea/blaze/golang/sync/BlazeGoAdditionalLibraryRootsProvider.java

```diff
@@ -71,6 +72,7 @@ protected String getLibraryName() {
     return BlazeGoPackage.getUncachedTargetToFileMap(project, projectData).values().stream()
         .filter(isExternal)
         .filter(f -> f.getName().endsWith(".go"))
+        .map(FileUtil::patchExternalFilePath)
```

Comment:
As you noticed, patching the file paths is necessary in several places in the code. All of the ones you modified get their data from `BlazeGoPackage#getUncachedTargetToFileMap`. Is there a reason you didn't directly patch the file paths in that method?

When I checked for usages, I found at least one additional code path which depends on it. In the current state of this PR, that code path wouldn't get patched file paths, which could lead to overall inconsistencies and thus further bugs.

Doing the file patching in `BlazeGoPackage#getUncachedTargetToFileMap` would additionally have the advantage that the patching wouldn't need to be done on every use of the targetToFileMap as `BlazeGoPackage#getTargetToFileMap` would then cache the patched version.